integration-tests/openapi-java/src/test/java/org/apache/camel/quarkus/component/openapijava/it/common/OpenApiTest.java

```diff
@@ -0,0 +1,224 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.openapijava.it.common;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.apache.camel.quarkus.component.openapijava.it.OpenApiContentType;
+import org.apache.camel.quarkus.component.openapijava.it.YamlToJsonFilter;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Tests common to both OpenAPI 2.x & 3.x
+ */
+public abstract class OpenApiTest {
+
+    @BeforeAll
+    public static void beforeAll() {
+        RestAssured.filters(new YamlToJsonFilter());
+    }
+
+    @Test
+    public void invokeApiEndpoint() {
+        RestAssured.given()
+                .queryParam("port", RestAssured.port)
+                .get("/api/fruits/list")
+                .then()
+                .contentType(ContentType.JSON)
+                .statusCode(200)
+                .body(
+                        "description", containsInAnyOrder("Winter fruit", "Tropical fruit"),
+                        "name", containsInAnyOrder("Apple", "Pineapple"));
+    }
+
+    @ParameterizedTest
+    @EnumSource(OpenApiContentType.class)
+    public void invokeApiDocumentEndpoint(OpenApiContentType contentType) {
+        RestAssured
+                .given()
+                .header("Accept", contentType.getMimeType())
+                .get("/openapi")
+                .then()
+                .contentType(ContentType.JSON)
+                .statusCode(200)
+                .body(
+                        "paths.'/api/fruits/list'", hasKey("get"),
+                        "paths.'/api/fruits/list'.get.summary", is("Gets a list of fruits"),
+                        "paths.'/api/fruits/list'.get.operationId", is("list"));
+    }
+
+    @ParameterizedTest
+    @EnumSource(OpenApiContentType.class)
+    public void openApiEndpointSecurity(OpenApiContentType contentType) {
+        RestAssured.given()
+                .header("Accept", contentType.getMimeType())
+                .get("/openapi")
+                .then()
+                .contentType(ContentType.JSON)
```

Comment:
We maybe need a clarification here.
So, we expect a response in YAML or JSON ? But the returned content type is always JSON ?

Reply:
The YAML response is intercepted & transformed to JSON so that we don't have to do custom verifcations on the response.

Reply:
Ok, now I remind the YamlToJsonFilter from previous review. Thanks @jamesnetherton .